### PR TITLE
allow phantom types

### PR DESCRIPTION
**Pull request by mlutze**
fixes https://github.com/flix/flix/issues/1550

### CODE REVIEW

**Comment by magnus-madsen** [comment]
!!!benchmark

File: main/test/flix/Test.Dec.OpaqueType.flix [hunk 1]
```diff
@@ -199,6 +199,20 @@ namespace Test/Dec/OpaqueType {
...
+
+    pub enum Metres {} // TODO make non-public
+
+    def metres(x: Float64): Distance[Metres] = Distance(x)
+
+    @test
+    def testOpaqueType34(): Bool = match metres(123.4) {
+        case Distance(d1) => d1 == 123.4
+    }
+
```
Comment (magnus-madsen) [inline main/test/flix/Test.Dec.OpaqueType.flix:215]: I don't really understand the problem or its fix. 

My guess is that this allows a (very) limited form of kind inference...

Would it not be possible to simply annotate `_unit` with its proper kind:

```
opaque type Distance[_unit :# <something here>] = Float64
```

?

Comment (magnus-madsen) [reply main/test/flix/Test.Dec.OpaqueType.flix:215]: What I mean-- is this not just a special case instance of the broader question of what to do about kinds and kind inference?

Comment (mlutze) [reply main/test/flix/Test.Dec.OpaqueType.flix:215]: Yes, I think that would be possible.

Right now we use `Kind.Var` where we can't immediately infer the kind. But we treat `Kind.Var` as `Kind.Unbound` (i.e., unifies with everything), no substitution produced. This change applies that pattern to Enums, where it wasn't used before.

So the kind question is definitely still there; this just extends our current protocol to a point we missed.

Comment (mlutze) [reply main/test/flix/Test.Dec.OpaqueType.flix:215]: Alternately, we could assume un-inferrable enum params to be `Star` type, which is what we do for class tparams.

I'll write up a "current state of kinds" issue for more clear discussion.

Comment (magnus-madsen) [reply main/test/flix/Test.Dec.OpaqueType.flix:215]: I think something like that would be more reasonable.

**Comment by magnus-madsen** [comment]
Can I close this (not merge) - pending a better solution?

**Comment by mlutze** [comment]
Yes, no problem.

**Pull request closed without merging**